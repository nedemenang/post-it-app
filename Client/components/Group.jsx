import React, {Component} from 'react';
import {
  BrowserRouter as Router,
  Route,
  Link
} from 'react-router-dom';
import '../public/style.css';
import $ from '../public/jquery.js';
import AppActions from '../actions/AppActions';
import {ListItem} from 'material-ui/List';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import getMuiTheme from 'material-ui/styles/getMuiTheme';
import {green100, green500, green700} from 'material-ui/styles/colors';
import AppStore from '../stores/AppStore';
import AppAPI from '../utils/appAPI';

const style = {margin: 5};
const muiTheme = getMuiTheme({
  palette: {
    primary1Color: green500,
    primary2Color: green700,
    primary3Color: green100,
  },
}, {
  avatar: {
    borderColor: null,
  },
  //userAgent: req.headers['user-agent'],
});



class Group extends Component {

groupClicked() {
   //console.log(this.props.group.groupId)
    AppAPI.getGroupMessages(this.props.group);
    AppAPI.getUsersNotInGroups(this.props.group);
    AppActions.selectGroup(this.props.group);
   // console.log(this.props.group);
  }
  
  constructor(props){
    super(props);
    this.groupClicked = this.groupClicked.bind(this);
    this.state= {};
  }
  render(){
    return(
      <MuiThemeProvider muiTheme={muiTheme}>
        <ListItem onClick={this.groupClicked}>
          <strong>{this.props.group.groupname}</strong>
        </ListItem>
      </MuiThemeProvider>
    );
  }
}

export default Group;