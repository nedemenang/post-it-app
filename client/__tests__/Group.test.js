import Group from '../components/Group.jsx';
import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('the group item component', () => {
  let mountedComponent;
  const group = {
    groupId: '-KrAwfM16qbLOig_mSOc',
    groupName: 'Test',
    newMessage: false
  };

  const loggedInUser = {
    displayName: 'nnamisco',
    email: 'nedemenang@gmail.com',
    id: 'TUsUaucS5nb5kuOP6SpvTUcX9vD3',
    isAuthenticated: true,
    phoneNo: null,
    profilePic: '//postitappnnam.herokuapp.com/static/files/blank-profile-pic.png'
  };

  const groupItem = () => {
    if (!mountedComponent) {
      mountedComponent = shallow(<Group group = {group}
        loggedInUser = {loggedInUser} key = {1}/>);
    }
    return mountedComponent;
  };

  beforeEach(() => {
    mountedComponent = undefined;
  });

  it('should render', () => {
    const component = groupItem();
    expect(component).toBeDefined();
  });


  it('should render without throwing an error', () => {
    const component = groupItem();
    expect(component.contains(group.groupName)).toBe(true);
  });

  it('should match snapshot test', () => {
    const component = groupItem();
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });


  it('should recieve props', () => {
    expect(Object.keys(groupItem().props()).length).toBeGreaterThan(0);
  });
});
