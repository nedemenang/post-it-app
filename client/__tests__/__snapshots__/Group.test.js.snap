// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the group item component should match snapshot test 1`] = `
<div>
  <ListItem
    autoGenerateNestedIndicator={true}
    containerElement="span"
    disableKeyboardFocus={false}
    disabled={false}
    initiallyOpen={false}
    insetChildren={false}
    nestedItems={Array []}
    nestedLevel={0}
    onKeyboardFocus={[Function]}
    onMouseEnter={[Function]}
    onMouseLeave={[Function]}
    onNestedListToggle={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    onTouchTap={[Function]}
    open={null}
    primaryTogglesNestedList={false}
    secondaryTextLines={1}
  >
    <strong>
      Test
       
    </strong>
  </ListItem>
</div>
`;
